### DISCUSSION TITLE
Add option to make layout more compact

### DISCUSSION THREAD

**Discussion by petermmiller** [iSoron/uhabits#335] [Feature Requests]
I'd love to see three **_independent_** 'compact layout' options for the main 'habits and checkmarks' screen:
- **Compact rows:**  Currently my creaky S5 can show 10 habit rows in portrait; since I rarely use a long habit name, there ends up being a lot of wasted space.
  - _Proposal:_  Selection option for 'Standard/Comfortable' (current double-height rows), 'Cosy' (1.5 height rows) and 'Compact' (single-height rows).  _<name suggestions lifted from Gmail!>_
- **Compact columns:**  _Similarly_, my S5 can show 4 day columns in portrait.
  - _Proposal:_  Selection option for 'Standard/Comfortable' (current double-width rows), 'Cosy' (1.5 width rows) and 'Compact' (single-width rows).
- **Compact headers:**  Same as per _Compact rows_ for the Title/Add/Filter/Settings header - which I rarely use.
  - _Proposal:_  Selection option for 'Standard/Comfortable' (current double-height rows), 'Cosy' (1.5 height rows) and 'Compact' (single-height rows).

Cheers!  =^..^=

**Comment by iSoron** [comment]
Thank you for the suggestion, @petermmiller. In order to keep the app simple, I am trying to avoid adding too many options. If such a feature ever makes it into the app, it will probably be simply a toggle "make layout more compact", which would trigger several of your suggestions simultaneously, not three independent toggles.

Another thing to consider is that the rows and columns are currently as wide and tall as they are because the checkmarks themselves are touch targets. Making them too small increases the likelihood that the user presses the wrong one. According to the Material Design guidelines, touch targets should be at least 48x48dp, which is precisely the current checkmark size.
